### PR TITLE
PieChart empty state

### PR DESCRIPTION
**Pull request by felipecadavid**
## Description of changes

This PR adds empty state to thee PieChart component and enables changing the error fallback icon's color

image

## Checklist

Before merging to main:

- [ ] Tests
- [x] Manually tested in React apps
- [x] Changesets
- [x] Approved

### CODE REVIEW

File: .changeset/curvy-phones-thank.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@propeldata/ui-kit': patch
+---
+
+Fix PieChart empty state
```
Comment (markandrus) [inline .changeset/curvy-phones-thank.md:5]: What was the behavior before? Can you document it here?

File: packages/ui-kit/src/components/ErrorFallback/ErrorFallback.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const ErrorFallback = React.forwardRef<HTMLDivElement, ErrorFallbackProps
           style={{ ...style }}
           data-testid="error-fallback-container"
         >
-          <Icon />
+          <Icon color={color} />
```
Comment (markandrus) [inline packages/ui-kit/src/components/ErrorFallback/ErrorFallback.tsx:43]: Is this a bugfix? It should be included in the changeset.

File: .changeset/curvy-phones-thank.md (hunk 1 above)
Comment (markandrus) [reply .changeset/curvy-phones-thank.md:5]: Also, can you define what the empty state is? Is it when there is no data?

**Review by markandrus — COMMENTED** [review]
Should we have a fallback that the user can customize, like we are planning to do for errors, loading, etc.?

**Comment by felipecadavid** [comment]
> Should we have a fallback that the user can customize, like we are planning to do for errors, loading, etc.?

Yes! I plan to do that in another PR though since that is more a new feature and this is a bug fix

File: .changeset/curvy-phones-thank.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@propeldata/ui-kit': patch
+---
+
+Fix PieChart empty state when no data is received from blank to a circle
+Fix ErrorFallback icon color prop not being passed
```
Comment (markandrus) [inline .changeset/curvy-phones-thank.md:6]:
```suggestion
- Fix PieChart empty state when no data is received from blank to a circle
- Fix ErrorFallback icon color prop not being passed
```

**Review by markandrus — APPROVED** [review]
Final suggestion; otherwise, +1

File: packages/ui-kit/src/components/PieChart/PieChart.stories.tsx [hunk 4]
```diff
@@ -314,3 +315,21 @@ export const ThemeStory: Story = {
...
 }
+
+export const EmptyPieStory: Story = {
+  name: 'Empty State',
+  args: {
+    query: {
+      ...connectedParams,
+      filters: [
+        {
+          column: 'tortilla_id',
```
Comment (sashathor) [inline packages/ui-kit/src/components/PieChart/PieChart.stories.tsx:326]: On my side, it fails with
```
{
    "data": {
        "leaderboard": null
    },
    "errors": [
        {
            "message": "Unknown column \"tortilla_id\"",
            "extensions": {
                "code": "UNKNOWN_COLUMN"
            }
        }
    ]
}
```

Shall we put it into env maybe?

File: packages/ui-kit/src/components/PieChart/plugins/empty.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { Chart } from 'chart.js'
+
+export const emptyPlugin = {
```
Comment (sashathor) [inline packages/ui-kit/src/components/PieChart/plugins/empty.ts:3]: `emptyPlugin` as a name is a bit confusing. `emptyStatePlugin` maybe?

File: packages/ui-kit/src/components/PieChart/plugins/empty.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    datasets.forEach((dataset) => {
+      if ((dataset.data.length ?? 0) === 0) {
+        hasData = false
+        return
+      }
+    })
+
+    if (!hasData) {
+      console.log('drawing')
```
Comment (sashathor) [inline packages/ui-kit/src/components/PieChart/plugins/empty.ts:22]: Please clean up the logs

File: packages/ui-kit/src/components/PieChart/PieChart.stories.tsx (hunk 4 above)
Comment (felipecadavid) [reply packages/ui-kit/src/components/PieChart/PieChart.stories.tsx:326]: Probably you don't have that dimension in your metric, let me set as an env variable so you can use any other

File: app/storybook/.env.example [hunk 7]
```diff
@@ -5,7 +5,8 @@ STORYBOOK_METRIC_UNIQUE_NAME_1="Revenue"
 STORYBOOK_DIMENSION_1="sauce_name"
 STORYBOOK_DIMENSION_2="taco_name"
 STORYBOOK_DIMENSION_3="restaurant_name"
+STORYBOOK_DIMENSION_4="tortilla_id"
```
Comment (sashathor) [inline app/storybook/.env.example:8]: 👍

**Review by sashathor — APPROVED** [review]
LGTM! 👍

**Pull request merged**